lightning/src/ln/channel.rs

```diff
@@ -1366,8 +1388,11 @@ impl<Signer: Sign> Channel<Signer> {
 			if htlc.htlc_id == htlc_id_arg {
 				match htlc.state {
 					InboundHTLCState::Committed => {},
-					InboundHTLCState::LocalRemoved(_) => {
-						debug_assert!(false, "Tried to fail an HTLC that was already fail/fulfilled");
+					InboundHTLCState::LocalRemoved(ref reason) => {
+						if let &InboundHTLCRemovalReason::Fulfill(_) = reason {
+						} else {
+							debug_assert!(false, "Tried to fail an HTLC that was already failed");
+						}
 						return Ok(None);
```

Comment:
The method-level comment says that we return a `ChannelError::Ignore` if we fail twice, but this seems to return `Ok(None)`?